Makefile

```diff
@@ -241,10 +241,13 @@ test:
 	cd tests && cargo test --no-default-features --features "${ENABLE_FEATURES}" ${EXTRA_CARGO_ARGS} -- --nocapture
 
 # This is used for CI test
-ci_test:
+ci_test: doc_test
 	cargo test --no-default-features --features "${ENABLE_FEATURES}" --all --exclude tests --all-targets --no-run --message-format=json
 	cd tests && cargo test --no-default-features --features "${ENABLE_FEATURES}" --no-run --message-format=json
 
+doc_test:
+	cargo test --no-default-features --features "${ENABLE_FEATURES}" --all --exclude tests --doc
```

Comment:
1. Can we call it ci_doc_test? Since only CI uses it and routine developement already covers it in other rules.
2. Maybe (not sure) `--exclude tests` can be removed.

@kennytm @nrc Do we have ways to grab the executable file that doc test runs? In this way, we can run doc tests like other tests (i.e. separated into two steps, 1. compile, 2. distribute test files into multiple nodes, 3. run different test partials in parallel).

Reply:
@breeswish Perhaps you could abuse rust-lang/rust#64245 to save the generated exe. Note that one exe is generated for each doc test, so if there are 100 doc tests there will be 100 exes.

Reply:
Doc test runs very quickly current time. @breeswish

Reply:
@mahjonp So maybe we can just run the doc test at compiling stage, after compiling test artifacts?